crates/rover-client/src/operations/subgraph/check_workflow/runner.rs

```diff
@@ -0,0 +1,149 @@
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/subgraph/check_workflow/check_workflow_query.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize, Clone",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. subgraph_check_workflow_query
+pub(crate) struct SubgraphCheckWorkflowQuery;
+
+/// The main function to be used from this module.
+/// This function takes a proposed schema and validates it against a published
+/// schema.
+pub fn run(
+    input: CheckWorkflowInput,
+    subgraph: String,
+    client: &StudioClient,
+) -> Result<CheckResponse, RoverClientError> {
+    let graph_ref = input.graph_ref.clone();
+    let mut data;
+    let now = Instant::now();
+    // default timeout is 5 minutes
+    let timeout_seconds = option_env!("APOLLO_CHECKS_TIMEOUT_SECONDS")
+        .unwrap_or_else(|| "300")
+        .parse::<u64>()
+        .unwrap();
+    loop {
+        data = client.post::<SubgraphCheckWorkflowQuery>(input.clone().into())?;
+        let graph = data.clone().graph.ok_or(RoverClientError::GraphNotFound {
+            graph_ref: graph_ref.clone(),
+        })?;
+        if let Some(check_workflow) = graph.check_workflow {
+            if !matches!(check_workflow.status, CheckWorkflowStatus::PENDING) {
+                break;
+            }
+        }
+        if now.elapsed() > Duration::from_secs(timeout_seconds) {
+            // TODO timeout error
+            eprintln!(
+                "Timeout after {} seconds waiting for check to complete, check again later.",
+                timeout_seconds
+            );
+            break;
+        }
+        std::thread::sleep(Duration::from_secs(5));
+    }
+    get_check_response_from_data(data, graph_ref, subgraph)
+}
+
+fn get_check_response_from_data(
+    data: QueryResponseData,
+    graph_ref: GraphRef,
+    subgraph: String,
+) -> Result<CheckResponse, RoverClientError> {
+    let graph = data.graph.ok_or(RoverClientError::GraphNotFound {
+        graph_ref: graph_ref.clone(),
+    })?;
+    let check_workflow = graph
+        .check_workflow
+        .ok_or(RoverClientError::GraphNotFound {
+            graph_ref: graph_ref.clone(),
+        })?;
+
+    let mut operations_result = None;
+    let mut target_url = None;
+    let mut status = ChangeSeverity::FAIL;
+    let mut number_of_checked_operations: u64 = 0;
+    let mut core_schema_modified = false;
+    let mut composition_errors = Vec::new();
+    for task in check_workflow.tasks {
+        match task.on {
+            OperationsCheckTask(typed_task) => {
+                target_url = task.target_url;
+                status = task.status.into();
```

Comment:
@Y-Guo @EverlastingBugstopper 
I think it would be better to get the check's status from `CheckWorkflow.status` instead of recomputing it in the client, as this handles tasks Rover doesn't know about better.

In particular, Studio is going to add a new `DownstreamCheckTask` for federated variants that can potentially fail a check. If you recompute the workflow status in Rover, it may disagree with the workflow status computed by Studio (so e.g. CI/CD could mistakenly think a check succeeded when it failed).

Also, I'm trying to think about how Rover behavior could degrade gracefully in the presence of task kinds it doesn't understand, and I'm curious -- would it be useful to add a field like this?
```graphql
extend interface CheckWorkflowTask {
  statusDetails: String!
}
```
The idea is if you encounter a type of task you don't understand and its `CheckWorkflowTask.status` is failing, you can fallback to just printing the `CheckWorkflowTask.statusDetails` string instead of just ignoring the task (which would be confusing for users).